clients/src/test/java/org/apache/kafka/clients/consumer/internals/HeartbeatRequestManagerTest.java

```diff
@@ -0,0 +1,346 @@
+import static org.apache.kafka.clients.consumer.internals.MemberState.FAILED;
+import static org.apache.kafka.clients.consumer.internals.MemberState.STABLE;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class HeartbeatRequestManagerTest {
+
+    private final int heartbeatIntervalMs = 1000;
+    private final long retryBackoffMaxMs = 3000;
+    private final long retryBackoffMs = 100;
+    private final String groupId = "group-id";
+
+    private Time mockTime;
+    private LogContext mockLogContext;
+    private CoordinatorRequestManager mockCoordinatorRequestManager;
+    private SubscriptionState mockSubscriptionState;
+    private HeartbeatRequestManager heartbeatRequestManager;
+    private MembershipManager mockMembershipManager;
+    private HeartbeatRequestManager.HeartbeatRequestState heartbeatRequestState;
+    private ConsumerConfig config;
+
+    private String memberId = "member-id";
+    private int memberEpoch = 1;
+    private ErrorEventHandler errorEventHandler;
+
+    private ConsumerGroupHeartbeatResponseData.Assignment mockAssignment() {
+        return new ConsumerGroupHeartbeatResponseData.Assignment()
+            .setAssignedTopicPartitions(Arrays.asList(
+                new ConsumerGroupHeartbeatResponseData.TopicPartitions()
+                    .setTopicId(Uuid.randomUuid())
+                    .setPartitions(Arrays.asList(0, 1, 2)),
+                new ConsumerGroupHeartbeatResponseData.TopicPartitions()
+                    .setTopicId(Uuid.randomUuid())
+                    .setPartitions(Arrays.asList(3, 4, 5))
+            ));
+    }
+
+    @BeforeEach
+    public void setUp() {
+        mockTime = new MockTime();
+        mockLogContext = new LogContext();
+        Properties properties = new Properties();
+        properties.put(BOOTSTRAP_SERVERS_CONFIG, "localhost:9999");
+        properties.put(KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class);
+        properties.put(VALUE_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class);
+        properties.put(RETRY_BACKOFF_MS_CONFIG, "100");
+        config = new ConsumerConfig(properties);
+        mockCoordinatorRequestManager = mock(CoordinatorRequestManager.class);
+        when(mockCoordinatorRequestManager.coordinator()).thenReturn(Optional.of(new Node(1, "localhost", 9999)));
+        mockSubscriptionState = mock(SubscriptionState.class);
+        mockMembershipManager = spy(new MembershipManagerImpl(groupId));
+        heartbeatRequestState = mock(HeartbeatRequestManager.HeartbeatRequestState.class);
```

Comment:
So by default `heartbeatRequestState` is a mock but we don't prefix it with `mock`. This is inconsistent...

Reply:
Thanks, i'll just get rid of the mock there.  Probably unnecessary.